src/engraving/dom/page.cpp

```diff
@@ -348,52 +394,61 @@ String Page::replaceTextMacros(const String& s) const
             {
                 int no = static_cast<int>(m_no) + 1 + score()->pageNumberOffset();
                 if (no > 0) {
-                    d += String::number(no);
+                    TextFragment pageNumberFragment(String::number(no));
+                    pageNumberFragment.format.setStyle(style().styleV(Sid::pageNumberFontStyle).value<FontStyle>());
+                    pageNumberFragment.format.setFontSize(style().styleD(Sid::pageNumberFontSize));
+                    pageNumberFragment.format.setFontFamily(style().styleSt(Sid::pageNumberFontFace));
+                    fragments.emplace_back(pageNumberFragment);
+                    fragments.emplace_back(TextFragment()); // Start next fragment
```

Comment:
You could make this conditional too: if the whole text only consists of the page number, i.e. the user has typed only `$p` or `$N` or `$P`, the entire TextBase should use page number style, so there is no need to create a separate TextFragment here.
(same for copyright)